concent_api/concent_api/middleware.py

```diff
@@ -39,3 +45,33 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+class HandleServerErrorMiddleware(object):
+    """
+    Used to catch all unhandled exceptions and return JSON response containing error information.
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.sid = None
+
+    def __call__(self, request):
+        self.sid = transaction.savepoint()
+        response = self.get_response(request)
+        transaction.savepoint_commit(self.sid)
+        return response
+
+    def process_exception(self, request, exception):  # pylint: disable=unused-argument
+        transaction.savepoint_rollback(self.sid)
+        logging.logger.exception(exception)
```

Comment:
Does your middleware bypass the built-in mechanism that reports HTTP 500 errors?
- If not then this logging is not necessary.
- If yes then you're missing the part that sends an error notification (by logging using the `mail_admins` handler). In effect we'll stop receiving error notifications via Sentry an e-mail.

Please compare the logging output when your middleware is enabled and when it's disabled and make sure that we're not losing any messages.

Reply:
ok, from my analysis, it seems that having my middleware put on top we should be safe with all logging handlers.
I will remove logging from my middleware

Reply:
done

Reply:
Looks like it does not work that way. The middleware is silencing the error notifications.